jupyterlab_automl/src/components/list_resources_panel.tsx

```diff
@@ -121,6 +133,153 @@ export class ListResourcesPanel extends React.Component<Props, State> {
     }
   }
 
+  private getTableForResourceType(type: ResourceType) {
+    const sharedProps = {
+      isLoading: this.state.isLoading,
+      height: this.props.height - 80,
+      width: this.props.width,
+    };
+    switch (type) {
```

Comment:
This is a pretty big switch statement. It looks like there might be some redundancy into the ListResourcesTable components that are returned from each block. Any chance the inner sections could be factored into other private methods so this method doesn't end up spanning so long? I think one of the rules of thumb that some of the Google style guides suggest is that if the method requires scrolling, it's a good candidate to split apart.